includes/Classifai/Providers/AWS/AmazonPolly.php

```diff
@@ -0,0 +1,595 @@
+	 *
+	 * @param int $post_id Post ID.
+	 * @return string|WP_Error
+	 */
+	public function synthesize_speech( int $post_id ) {
+		if ( empty( $post_id ) ) {
+			return new WP_Error(
+				'aws_polly_post_id_missing',
+				esc_html__( 'Post ID missing.', 'classifai' )
+			);
+		}
+
+		// We skip the user cap check if running under WP-CLI.
+		if ( ! current_user_can( 'edit_post', $post_id ) && ( ! defined( 'WP_CLI' ) || ! WP_CLI ) ) {
+			return new WP_Error(
+				'aws_polly_user_not_authorized',
+				esc_html__( 'Unauthorized user.', 'classifai' )
+			);
+		}
+
+		$normalizer          = new Normalizer();
+		$feature             = new TextToSpeech();
+		$settings            = $feature->get_settings();
+		$post                = get_post( $post_id );
+		$post_content        = $normalizer->normalize_content( $post->post_content, $post->post_title, $post_id );
+		$content_hash        = get_post_meta( $post_id, self::AUDIO_HASH_KEY, true );
+		$saved_attachment_id = (int) get_post_meta( $post_id, $feature::AUDIO_ID_KEY, true );
+
+		// Don't regenerate the audio file it it already exists and the content hasn't changed.
+		if ( $saved_attachment_id ) {
+
+			// Check if the audio file exists.
+			$audio_attachment_url = wp_get_attachment_url( $saved_attachment_id );
+
+			if ( $audio_attachment_url && ! empty( $content_hash ) && ( md5( $post_content ) === $content_hash ) ) {
+				return $saved_attachment_id;
+			}
+		}
+
+		if ( mb_strlen( $post_content ) > 3000 ) {
+			return new WP_Error(
+				'aws_polly_length_error',
+				esc_html__( 'Maximum text length has been exceeded.', 'classifai' )
+			);
+		}
+
+		$voice = $settings[ static::ID ]['voice'] ?? '';
+
+		try {
+			/**
+			 * Filter Synthesize speech args.
+			 *
+			 * @since 3.1.0
+			 * @hook classifai_aws_polly_synthesize_speech_args
+			 *
+			 * @param {array} Associative array of synthesize speech args.
+			 * @param {int}   $post_id Post ID.
+			 * @param {object} $provider_instance Provider instance.
+			 * @param {object} $feature_instance Feature instance.
+			 *
+			 * @return {array} The filtered array of synthesize speech args.
+			 */
+			$synthesize_data = apply_filters(
+				'classifai_' . self::ID . '_synthesize_speech_args',
+				array(
+					'OutputFormat' => 'mp3',
+					'Text'         => $post_content,
+					'TextType'     => 'text',
+					'Engine'       => $settings[ static::ID ]['voice_engine'] ?? 'standard',
+					'VoiceId'      => $voice,
+				),
+				$post_id,
+				$this,
+				$this->feature_instance
+			);
+
+			/**
+			 * Filters the return value of the synthesize speech function.
+			 *
+			 * Returning a non-false value from the filter will short-circuit the synthesize speech request and return early with that value.
+			 * This filter is useful for E2E tests.
+			 *
+			 * @since 3.1.0
+			 *
+			 * @param false|mixed $response        A return value of synthesize speech. Default false.
+			 * @param array       $synthesize_data HTTP request arguments.
+			 */
```

Comment:
We'll want to ensure this docblock matches what JSDoc expects so it ends up in our documentation. See the `classifai_aws_polly_synthesize_speech_args` filter for an example